update logic of adding default DS pod tolerations

Huang-Wei opened the pull request:
**What this PR does / why we need it**:

- update DS pod default tolerations: add PIDPressure, remove OutOfDisk
- remove useless tolerations testcases

**Which issue(s) this PR fixes**:

- Fixes partial of comment https://github.com/kubernetes/kubernetes/issues/66348#issuecomment-410419351.

**Special notes for your reviewer**:

Correct the behavior of daemonset controller as "out-of-disk" toleration doesn't fit any more.

**Release note**:
```release-note
`node.kubernetes.io/pid-pressure` toleration is added for DaemonSet pods, and `node.kubernetes.io/out-of-disk` isn't added any more even if it's a critical pod.
```

/sig scheduling
/sig apps

Huang-Wei commented:
/assign @mikedanese @k82cn

MHBauer commented:
/ok-to-test

Huang-Wei commented:
/retest

Huang-Wei commented:
Friendly ping @k82cn @mikedanese :)

Huang-Wei commented:
@janetkuo @mikedanese Could you kindly review this PR?

It's a (small) behavior fix on default daemonset pods:

- remove OutOfDisk toleration (as OutOfDisk condition/taint has been removed in sig/node side for more than 2 releases)
- add PIDPressure toleration

k82cn commented:
/lgtm

k82cn commented:
ping @mikedanese , PTAL :)

Huang-Wei commented:
/retest

Huang-Wei commented:
/retest

Huang-Wei commented:
Friendly ping @mikedanese @janetkuo PTAL

k82cn commented:
@janetkuo , I think we'd better to get this merged in 1.12 :)

Huang-Wei commented:
@k82cn thanks for bringing this up.
This PR is very likely to need rebase (due to PR #64954) and *minor* code changes. If @janetkuo thinks it's a 1.12 candidate, I will rebase this one as soon as #64954 is merged.

Huang-Wei commented:
Code rebased. @k82cn PTAL.

k82cn commented on pkg/controller/daemon/util/daemonset_util.go:
```diff
@@ -120,11 +108,7 @@ func AddOrUpdateDaemonPodTolerations(spec *v1.PodSpec, isCritical bool) {
 func CreatePodTemplate(ns string, template v1.PodTemplateSpec, generation *int64, hash string) v1.PodTemplateSpec {
```
also remove `ns string` here :)

Huang-Wei replied:
I did see a reminder in one of the comments, but forgot...

Done.

Huang-Wei commented:
/retest

Huang-Wei commented:
/retest

Huang-Wei commented:
/retest

k82cn commented:
LGTM, for 1.13, please help to rebase :)

Huang-Wei commented:
@mikedanese @janetkuo could you kindly take a look at the Daemonset part?

Huang-Wei commented:
Friendly ping @mikedanese @janetkuo :)

k82cn commented:
@Huang-Wei , would you help to rebase this PR? I'd like to get this merged in 1.14 :)

Huang-Wei committed:
update logic of adding default DS pod tolerations

- update DS pod default tolerations: add PIDPressure, remove OutOfDisk
- remove useless tolerations testcases

Huang-Wei commented:
@k82cn rebased. PTAL.

Huang-Wei commented:
/retest

k82cn commented:
/lgtm
/approve

We're going to remove `OutOfDisk` in #67037 :)
